slides/linear_model.md

```diff
@@ -0,0 +1,260 @@
+* How to avoid overfitting?
+
+
+---
+# Adult census
+
+.very-small[
+
+| Age | Workclass | Education    | Marital-status     | Occupation         | Relationship | Race  | Sex  | Capital-gain | Hours-per-week | Native-country | Salary |
+| --- | --------- | ------------ | ------------------ | ------------------ | ------------ | ----- | ---- | ------------ | -------------- | -------------- | ----- |
+| 25  | Private   | 11th         | Never-married      | Machine-op-inspct  | Own-child    | Black | Male | 0            | 40             | United-States  | $45k |
+| 38  | Private   | HS-grad      | Married-civ-spouse | Farming-fishing    | Husband     | White  | Male | 0            | 50             | United-States   | $40k |
+| 28  | Local-gov | Assoc-acdm   | Married-civ-spouse | Protective-serv    | Husband      | White | Male | 0            | 40             | United-States   | $60k  |
+| 44  | Private   | Some-college | Married-civ-spouse | Machine-op-inspct  | Husband      | Black | Male | 7688         | 40             | United-States   | $52k  |
+
+]
+Salary = *.4 x* education + *.2 x* hours-per-week + *.1 x* age + ...
+
+???
+Adult census is here a bit modify, instead we have the value of the salary for every one.
+So we can see it as a regression problem
+
+Salary could be a linear combination of the feature (explanatory variable). 
+
+
+
+---
+# Linear regression
+
+Predict the value of the target **y**  
+given some observation **X**
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_data.svg" width="100%">]
+
+???
+For illustration purpose, let's consider one dimensionnal observation, 
+e.g. salary should be explain by education level (nb of year of study)
+
+---
+# Linear regression
+A linear model is a slope "as close as possible" from datapoint
+The blue curve is the prediction for each **x**
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_fit.svg" width="100%">]
+.pull-right.very-small[```python
+from sklearn.linear_model import LinearRegression
+lin_reg = LinearRegression()
+lin_reg.fit(x, y)
+```]
+???
+
+We learn a linear function to predict *y*. Here the salary is a constant times the number of years of study.
+
+
+
+---
+# Linear regression
+
+The linear regression objectif is to minimize the distance between the prediction curve and the datapoints
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_fit_red.svg" width="100%">]
+
+???
+An error for each point correspond to the red bar in the figure
+the best fit is the line which minimize the sum of (the square of) those red lines.
+
+Fortunatly, there is a formula, given **X** and **y**, to find the optimal weights in an efficient manner
+
+---
+# Linear regression in higher dimension
+
+.pull-left.shift-left[<img src="../figures/lin_reg_3D.svg" width="130%">]
+
+
+???
+In real world data set, **X** has severals dimensions, and it is not possible anymore to represent it.
+
+---
+# For classification: logistic regression
```

Comment:
It would be great if it would fit on a single line

Reply:
Which line ? 
The title in my browser fit on a single line

Reply:
Yes, the title. So it could be a browser/system dependent.
We should be probably taking care of it to ensure that we have the same rendering everywhere but this might not be the highest priority right now. We can make it as a FIXME for later